packages/aragon-wrapper/src/index.js

```diff
@@ -1507,19 +1506,21 @@ export default class Aragon {
       ]
     })
 
-    // Find the shortest path
+    // Find the shortest path via a depth-first search of forwarder paths
+    // We eagerly evaluate paths that look promising; as soon as we find a forwarder that
+    // "links" up with a previous forwarder, we continue exploring this path immediately.
     // TODO(onbjerg): Should we find and return multiple paths?
     do {
       const [path, [forwarder, ...nextQueue]] = queue.shift()
 
-      // Skip paths longer than 10
-      if (path.length > 10) continue
+      // Skip if no forwarder or the path is longer than 5
+      if (!forwarder || path.length > 5) continue
```

Comment:
The `!forwarder` check was the important one we were missing before; even though `nextQueue` would be empty, we'd still treat it as a normal path and re-add it back to the path at the end.